app/test/unit/git/reflog-test.ts

```diff
@@ -29,42 +31,71 @@ describe('git/reflog', () => {
+      await createAndCheckout(repository, 'branch-2')
 
-      const branches = await getRecentBranches(repository!, 10)
+      const branches = await getRecentBranches(repository, 10)
       expect(branches).toContain('branch-1')
       expect(branches).toContain('branch-2')
     })
 
     it('works after renaming a branch', async () => {
-      await createAndCheckout(repository!, 'branch-1')
-      await createAndCheckout(repository!, 'branch-2')
+      await createAndCheckout(repository, 'branch-1')
+      await createAndCheckout(repository, 'branch-2')
 
-      const allBranches = await getBranches(repository!)
+      const allBranches = await getBranches(repository)
       const currentBranch = allBranches.find(
         branch => branch.name === 'branch-2'
       )
 
-      await renameBranch(repository!, currentBranch!, 'branch-2-test')
+      await renameBranch(repository, currentBranch!, 'branch-2-test')
 
-      const branches = await getRecentBranches(repository!, 10)
+      const branches = await getRecentBranches(repository, 10)
       expect(branches).not.toContain('master')
       expect(branches).not.toContain('branch-2')
       expect(branches).toContain('branch-1')
       expect(branches).toContain('branch-2-test')
     })
 
     it('returns a limited number of branches', async () => {
-      await createAndCheckout(repository!, 'branch-1')
-      await createAndCheckout(repository!, 'branch-2')
-      await createAndCheckout(repository!, 'branch-3')
-      await createAndCheckout(repository!, 'branch-4')
+      await createAndCheckout(repository, 'branch-1')
+      await createAndCheckout(repository, 'branch-2')
+      await createAndCheckout(repository, 'branch-3')
+      await createAndCheckout(repository, 'branch-4')
 
-      const branches = await getRecentBranches(repository!, 2)
+      const branches = await getRecentBranches(repository, 2)
       expect(branches).toHaveLength(2)
       expect(branches).toContain('branch-4')
       expect(branches).toContain('branch-3')
     })
   })
+
+  describe('getCheckoutsAfterDate', () => {
+    it('returns does not return the branches that were checked out before a specific date', async () => {
+      await createAndCheckout(repository!, 'branch-1')
+      await createAndCheckout(repository!, 'branch-2')
+
+      const branches = await getCheckoutsAfterDate(
+        repository!,
+        moment()
+          .add(1, 'day')
+          .toDate()
+      )
+      expect(branches.size).toBe(0)
+    })
+
+    it('returns all branches checked out after a specific date', async () => {
+      await createBranch(repository!, 'never-checked-out')
+      await createAndCheckout(repository!, 'branch-1')
+      await createAndCheckout(repository!, 'branch-2')
+
+      const branches = await getCheckoutsAfterDate(
+        repository!,
+        moment()
+          .subtract(1, 'hour')
+          .toDate()
+      )
+      expect(branches.size).toBe(2)
```

Comment:
💭  we don't seem to be doing any checks for the `Date` returned with each branch. Is it worth doing any tests for that in here?

Reply:
I didn't feel like it would add any value since I'm checking out the branch as a part of the test. If this were a test fixture, it would be worth it because we could use it to potentially catch regressions. Adding the time component wont give us that benefit since the date would change each time the test is run.

Reply:
What about opening an issue to track testing a repository with known checkout times?

Reply:
Issues created here: https://github.com/desktop/desktop/issues/6855